spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -210,6 +211,16 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		return taskExecution.getExecutionId();
 	}
 
+	private boolean commandLineArgsContainNoDataSourceInfo(List<String> commandLineArgs) {
+		boolean result = true;
+		for(String commandLineArg : commandLineArgs) {
+			if(commandLineArg.startsWith("--spring.datasource.")) {
```

Comment:
What happens if the user passes one of the `spring.datasource.*` properties and expect the other properties to be injected as usual? Currently, if the user explicitly set some of the `spring.datasource.*` properties, then we pass only those properties.